File: app/src/main/java/co/tpcreative/supersafe/common/preference/MyPreferenceAlbumSettings.java
package co.tpcreative.supersafe.common.preference;

import android.content.Context;
import android.support.v7.preference.Preference;
import android.support.v7.preference.PreferenceViewHolder;
import android.util.AttributeSet;
import android.widget.ImageView;

import co.tpcreative.supersafe.R;
import co.tpcreative.supersafe.common.util.Utils;


public class MyPreferenceAlbumSettings extends Preference {

    private static final String TAG = MyPreferenceAlbumSettings.class.getSimpleName();
    private Context context;
    private ImageView imageView;
    private ImageView imgIcon;
    private MyPreferenceListener listener;

    public void setListener(MyPreferenceListener listener) {
        this.listener = listener;
    }

    public MyPreferenceAlbumSettings(Context context) {
        super(context);
        this.context = context ;
    }

    public MyPreferenceAlbumSettings(Context context, AttributeSet attrs) {
        super(context, attrs);
        this.context = context;
    }

    public MyPreferenceAlbumSettings(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        this.context = context;
    }

    @Override
    public void onBindViewHolder(PreferenceViewHolder view) {
        super.onBindViewHolder(view);
        imageView = (ImageView) view.findViewById(R.id.imgCover);
        imgIcon = (ImageView) view.findViewById(R.id.imgIcon);
        if (listener!=null){
            listener.onUpdatePreference();
        }
        Utils.Log(TAG,"onBind....");
    }

    public ImageView getImageView() {
        return imageView;
    }

    public ImageView getImgIcon() {
        return imgIcon;
    }

    public interface MyPreferenceListener{
        void onUpdatePreference();
    }

}

File: app/src/main/java/co/tpcreative/supersafe/common/preference/MyPreferenceAlbumSettings.kt
package co.tpcreative.supersafe.common.preference
import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.ImageView
import androidx.preference.Preference
import androidx.preference.PreferenceViewHolder
import co.tpcreative.supersafe.R
import kotlinx.android.synthetic.main.custom_preferences_item_album_settings.view.*

class MyPreferenceAlbumSettings @JvmOverloads constructor(
        context: Context,
        attrs: AttributeSet,
        defStyleAttr: Int = 0
) : Preference(context, attrs, defStyleAttr) {
    init {
        widgetLayoutResource = R.layout.custom_preferences_item_album_settings
    }
    override fun onBindViewHolder(holder: PreferenceViewHolder) {
        super.onBindViewHolder(holder)
        with(holder.itemView) {
            // do the view initialization here...
            //textView.text = "Another Text"
            imageViewCover = this.imgCover
            imgViewSuperSafe = this.imgIcon
            imageViewCover?.visibility = View.INVISIBLE
            imgViewSuperSafe?.visibility = View.INVISIBLE
            onUpdatedView?.invoke()
        }
    }
    var imageViewCover : ImageView? = null
    var imgViewSuperSafe : ImageView? = null
    var onUpdatedView : (() -> Unit)? = null
}